src/librustdoc/passes/collect_intra_doc_links.rs

```diff
@@ -343,6 +346,7 @@ impl ItemFragment {
 
 #[derive(Clone, Debug, Hash, PartialEq, Eq)]
 struct ResolutionInfo {
+    item_id: ItemId,
     module_id: DefId,
```

Comment:
Just wondering, could we remove `module_id` now that we have `item_id`? Not necessary to address now, but I wanted to bring it up.

Reply:
Not without replacing it with some extra data, I tried it but wasn't able to do it right away.

The first issue is that rustdoc (unlike rustc) resolves paths differently in inner and outer attributes.
So `item_id` will be the same for inner and outer paths, but their `module_id`s will be different.

There are also may be nuances with how `item_id` and `module_id` are assigned for reexports, but I didn't investigate it.

I agree that ideally we would have an `item_id` and some additional orthogonal flags here instead of the `module_id`.